src/main/java/org/owasp/benchmark/helpers/PropertiesManager.java

```diff
@@ -0,0 +1,123 @@
+package org.owasp.benchmark.helpers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class PropertiesManager {
+	private String propertiesFileName = null;
+	private File file = null;
+	private boolean isExternalFile = false;
+
+	public PropertiesManager() {
+		propertiesFileName = "benchmark.properties";
+		file = Utils.getFileFromClasspath(propertiesFileName, this.getClass().getClassLoader());
+	}
+
+	public PropertiesManager(String fileName) {
+		propertiesFileName = fileName;
+		file = Utils.getFileFromClasspath(propertiesFileName, this.getClass().getClassLoader());
+	}
+
+	public PropertiesManager(String path, String fileName) {
+		isExternalFile = true;
+		propertiesFileName = fileName;
+		file = new File(path + File.separator + fileName);
+		if (!file.exists()) {
+			try {
+				file.createNewFile();
+			} catch (IOException e) {
+				System.out.println("Problem creating new properties file.");
+			}
+		}
+	}
+
+	public void displayProperties() {
+		Properties props = new Properties();
+		InputStream is = null;
+		try {
+			is = this.getClass().getClassLoader().getResourceAsStream(propertiesFileName);
+			props.load(is);
+		} catch (Exception e) {
+		}
+
+		System.out.println(props.keySet());
+		System.out.println(props.values());
+	}
+
+	public String getProperty(String key, String defaultValue) {
+		Properties props = new Properties();
+		InputStream is = null;
+		try {
+			if (isExternalFile) {
+				is = new FileInputStream(file);
+			} else {
+				is = this.getClass().getClassLoader().getResourceAsStream(propertiesFileName);
+			}
+			props.load(is);
+		} catch (Exception e) {
+		}
+
+		return props.getProperty(key, defaultValue);
+	}
+
+	public int getProperty(String key, int defaultValue) {
+		Properties props = new Properties();
+		InputStream is = null;
+		try {
+			if (isExternalFile) {
+				is = new FileInputStream(file);
+			} else {
+				is = this.getClass().getClassLoader().getResourceAsStream(propertiesFileName);
+			}
+			props.load(is);
+		} catch (Exception e) {
+		}
+
+		return Integer.parseInt(props.getProperty(key, Integer.toString(defaultValue)));
+	}
+
+	public void saveProperty(String key, String value) {
+		InputStream in = null;
+		try {
+			if (isExternalFile) {
+				in = new FileInputStream(file);
+			} else {
+				in = this.getClass().getClassLoader().getResourceAsStream(propertiesFileName);
+			}
+			Properties props = new Properties();
+			props.load(in);
+			in.close();
+
+			FileOutputStream out = new FileOutputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *out*. The resource is referenced in statements at the following line: 97. The resource closure statement is at line: 98. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Properties.store. Either a) close *out* in a try-finally block or b) close the resource by declaring *out* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).